app/knights/knight.py

```diff
@@ -0,0 +1,37 @@
+from .equipment import Armor, Weapon, Potion
+
+
+class Knight:
+    def __init__(
+        self,
+        name: str,
+        base_power: int,
+        base_hp: int,
+        armor: list,

```

Comment:
Consider specifying the type of elements in the `armor` list for better clarity and type checking. For example, use `List[Armor]` instead of `list` if you are using Python 3.9 or later, or `list[Armor]` for earlier versions.